roles/add-sonar-plugin/tasks/main.yml

```diff
@@ -3,27 +3,41 @@
 - name: Get JAR name
   set_fact:
     jar_name: "{{ plugin_url.split('/')[-1] }}"
-
-- name: Download the SonarQube plugin
-  get_url:
-    url: "{{ plugin_url }}"
-    dest: "/tmp/{{ jar_name }}"
-    mode: 0440
+    plugin_name: "{{ plugin_url.split('/')[-1] | regex_replace('-([0-9]+\\.?)+\\.jar', '') }}"
 
 ## The Jinja2 template and the Go template in the following task conflict,
 ## so the weird double-curly-braces surrounded by double-curly-braces is required.
 - name: Find sonarqube pod
   command: >-
-      oc get pods --namespace={{ namespace }} -l="app==sonarqube"
+      oc get pods --namespace={{ namespace }} -l='app==sonarqube'
       -o jsonpath="{.items[0].metadata.name}"
-  register: pod_name_lines
+  register: pod_name
 
 - name: Extract pod name
   set_fact:
-    pod_name: "{{ pod_name_lines.stdout }}"
+    pod_name: "{{ pod_name.stdout }}"
+
+- name: Check for existing plugin install_location
+  command: >-
+    oc exec -i --namespace={{ namespace }} {{ pod_name }} -- find {{ install_location }}/ -type f -name '{{ plugin_name }}*jar'
+  register: existing_plugin
+
+- name: Download the SonarQube plugin
+  get_url:
+    url: "{{ plugin_url }}"
+    dest: "/tmp/{{ jar_name }}"
```

Comment:
Either use [tempfile](https://docs.ansible.com/ansible/2.4/tempfile_module.html), or make sure the permissions are set to something more open than `0440` - otherwise this will cause problems if different users are using the same environment to run this task